File: elfa-vz-verifier-ui/src/app/_models/attribute-group.ts
import {Attribute} from '@app/_models/attribute';

export class AttributeGroup {
  name: string;
  order?: number = null;
  attributes: Attribute[];

  constructor(name: string, order: number, attributes: Attribute[]) {
    this.name = name;
    this.order = order;
    this.attributes = attributes;
  }

  getAttribute(name: string) {
    return this.attributes.find(attribute => attribute.name === name);
  }
}

File: elfa-vz-verifier-scs/src/main/java/ch/admin/astra/elfa/trotti/domain/verifier/model/AttributeGroup.java
package ch.admin.astra.elfa.trotti.domain.verifier.model;


import lombok.Builder;
import lombok.Getter;
import lombok.Singular;
import lombok.extern.jackson.Jacksonized;

import java.util.List;

@Getter
@Builder
@Jacksonized
public final class AttributeGroup {

    private final String name;
    private final Long order;
    @Singular
    private final List<Attribute> attributes;
}
